tutorial/source/ekf.ipynb

```diff
@@ -0,0 +1,247 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import math\n",
+    "\n",
+    "import torch\n",
+    "import pyro\n",
+    "import pyro.distributions as dist\n",
+    "from pyro.contrib.autoguide import AutoDelta\n",
+    "from pyro.optim import Adam\n",
+    "from pyro.infer import SVI, Trace_ELBO, config_enumerate\n",
+    "from pyro.contrib.tracking.extended_kalman_filter import EKFState\n",
+    "from pyro.contrib.tracking.distributions import EKFDistribution\n",
+    "from pyro.contrib.tracking.dynamic_models import NcvContinuous\n",
+    "from pyro.contrib.tracking.measurements import PositionMeasurement\n",
+    "\n",
+    "smoke_test = ('CI' in os.environ)\n",
+    "pyro.enable_validation(True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dt = 1e-2\n",
+    "num_frames = 10\n",
+    "dim = 4\n",
+    "\n",
+    "# Continuous model\n",
+    "ncv = NcvContinuous(dimension=dim, sa2=2.0)\n",
+    "\n",
+    "# Truth trajectory\n",
+    "xs_truth = torch.zeros(num_frames, dim)\n",
+    "# initial direction\n",
+    "theta0_truth = 0.0\n",
+    "# initial state\n",
+    "xs_truth[0, :] = torch.tensor([0.0, 0.0,  math.cos(theta0_truth), math.sin(theta0_truth)])\n",
+    "for frame_num in range(1, num_frames):\n",
+    "    # sample independent process noise\n",
+    "    dx = pyro.sample('process_noise_{}'.format(frame_num), ncv.process_noise_dist(dt))\n",
+    "    xs_truth[frame_num, :] = ncv(xs_truth[frame_num-1, :], dt=dt) + dx"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let's specify the measurements. Notice that we only measure the positions of the particle."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Measurements\n",
+    "measurements = []\n",
+    "mean = torch.zeros(2)\n",
+    "# no correlations\n",
+    "cov = 1e-5 * torch.eye(2)\n",
+    "# sample independent measurement noise\n",
+    "dzs = pyro.sample('dzs', dist.MultivariateNormal(mean, cov).expand((num_frames,)))\n",
+    "# compute measurement means\n",
+    "zs = xs_truth[:, :2] + dzs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll use a [Delta autoguide](http://docs.pyro.ai/en/0.2.1-release/contrib.autoguide.html#autodelta) to learn the position and measurement covariances. The `EKFDistribution` computes the joint log density of all of the EKF states given a tensor of sequential measurements."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def model(data):\n",
+    "    # TODO replace this with HalfNormal\n",
```

Comment:
nit: remove TODO? `HalfCauchy` seems fine to me. Alternatively `Chi2` would be appropriate.

Reply:
halfcauchy seems too fat? i only used it because we dont have halfnormal in 0.4.0 but i had to shrink its width significantly. ill put a more aesthetic comment though